### PR TITLE
Update the PR testing workflow

### PR DESCRIPTION
**Pull request by dshevtsov**
## Purpose of this pull request

This pull request (PR) updates the PR testing GH workflow to split the testing jobs into three for better control and maintenance. Also, it removes code duplication using a composed action (https://github.com/commerce-docs/devsite-install-action) in all relevant workflows and removes unnecessary permissions in the testing PR workflow and the Pages workflow. Further restriction updates will be done later in separate PRs.

### CODE REVIEW

File: .github/workflows/test-pull-request.yml [hunk 1]
```diff
@@ -1,89 +1,122 @@
...
-      - name: Build site
-        run: yarn build
+    build:
+      runs-on: ubuntu-latest
+      needs: test
+      permissions:
+        contents: read
+        statuses: write
+      steps:
+        - name: Checkout Code
```
Comment (jeff-matthews) [inline .github/workflows/test-pull-request.yml:103]: Is it necessary to duplicate/repeat these steps? They've already been performed in the `test` job.

Comment (dshevtsov) [reply .github/workflows/test-pull-request.yml:103]: Every job is run from scratch by design.

Comment (jeff-matthews) [reply .github/workflows/test-pull-request.yml:103]: For what purpose?

Comment (dshevtsov) [reply .github/workflows/test-pull-request.yml:103]: By GitHub design.

**Review by jeff-matthews — APPROVED** [review]
Approved, but duplicating steps seems like an anti-pattern.

File: .github/workflows/test-pull-request.yml (hunk 1 above)
Comment (dshevtsov) [reply .github/workflows/test-pull-request.yml]: Found a way to avoid the duplication via a composite action.

Comment (jeff-matthews) [reply .github/workflows/test-pull-request.yml]: Nice work!

**Pull request merged**